src/fsharp/azure-pipelines.yml

```diff
@@ -154,15 +154,15 @@ extends:
                 testResultsFiles: '*.xml'
                 searchFolder: '$(Build.SourcesDirectory)/artifacts/TestResults/$(_BuildConfig)'
               continueOnError: true
-              condition: ne(variables['SkipTests'], 'true')
+              condition: and(succeeded(), ne(variables['SkipTests'], 'true'))
```

Comment:
this also sounds like we'd not publish test results if the previous step fails (which I assume it does if there are test failures)